### PR TITLE
:wrench: Improved Accessibility Permissions handling

### PR DESCRIPTION
**Pull request by JoniVR**
Small Quality of Life change.

~Currently when the app shows the popup asking for accessibility permissions, the app doesn't work right after granting these permissions because the app is still launched with previous permissions (would be my guess).~

~This just makes sure the app exits after asking for permissions and tells the user to restart manually after setting permissions. I think it's a bit more user friendly, as the app doesn't work without a manual restart after granting permissions, but let me know what you think.~

~I looked into somehow listening for this setting but I don't think it's possible.~

~Also, you might want to proofread those French and German translations, since I used [DeepL](https://www.deepl.com) for translations~

edit:
Permissions now work once the user grants them without having to manually restart the app.

### CODE REVIEW

File: MonitorControl/Support/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -38,5 +38,5 @@
 "Volume" = "Lautstärke";
 
 /* Shown in the alert dialog */
-"You need to enable MonitorControl in System Preferences > Security and Privacy > Accessibility for the keyboard shortcuts to work" = "Du musst MonitorControl in Systemeinstellungen > Sicherheit > Datenschutz > Bedienungshilfen aktivieren, damit die Kurzbefehle funktionieren.";
+"You need to enable MonitorControl in System Preferences > Security and Privacy > Accessibility for the keyboard shortcuts to work" = "Du musst MonitorControl in Systemeinstellungen > Sicherheit > Datenschutz > Bedienungshilfen aktivieren, damit die Kurzbefehle funktionieren.\n\nMonitorControl wird nun beendet. Bitte starten Sie nach Erteilung der Berechtigung manuell neu.";
```
Comment (reitermarkus) [inline MonitorControl/Support/de.lproj/Localizable.strings]: Bitte starte MonitorControl nach Erteilung der Berechtigung manuell neu.

**Comment by the0neyouseek** [comment]
I don't know if we should do this, making the app unusable unless the user does an action isn't really great in an usability stand point.  
Maybe we should simply put the message "Restart when done" like you did and add a button that restart the app on the menulet with a message (like chrome does for their updates). Something like ":warning: Accessibility permissions required. Click to restart when enabled"

**Comment by JoniVR** [comment]
I agree, that's a better solution.

**Comment by JoniVR** [comment]
example
Any suggestions for a shorter message? Or a different solution? I think this looks a bit too weird lol.

I was thinking maybe "⚠ Accessibility Permissions - Restart Now." but that doesn't really pass the message along well enough. Or just "⚠ Restart Now."

**Comment by the0neyouseek** [comment]
Can we detect if permissions were changed ? In that case it could be `Acessibility permissions required` at first, and after enabling`:warning: Permissions changed, restart now`
If not I guess simply `:warning: Restart app now` will do

**Comment by JoniVR** [comment]
> Can we detect if permissions were changed ?

I initially thought this was impossible, but after doing some deep digging I think I found out a way to do this. It's almost entirely undocumented but some apps (like Magnet) do this already. 

Apparently, you can just observe the System Preferences "com.apple.accessibility.api" distributed notification. [(source)](https://pfiddlesoft.com/accessibilitytestbench/)

like this:
```swift
DistributedNotificationCenter.default().addObserver(self, selector: #selector(self.readPrivileges), name: NSNotification.Name(rawValue: "com.apple.accessibility.api"), object: nil)
```

to detect a change in Accessibility permissions for your applications. 

Magnet.app even manages to not require a restart at all after changing permissions. 

So I'm trying to figure out how to do this too. This way we'd be able to show a warning with a button to set accessibility permissions in the menulet and not have to restart when the user grants the permission. Which I think would be most ideal.

You can easily test this on Magnet if you have it, if you revoke their permissions their menulet shows:
image

Once you authorize, it just works!

**Comment by JoniVR** [comment]
Ok so I figured out what caused the freezing (was something really stupid), right now it works pretty well. No restarting required at all, It just restarts mediakeytap and that's enough, then it just works, I don't know how I overlooked that.. 

So if the user grants permission it will automatically start working without restarting the app. The only thing I'd like to add before merging is that thing that Magnet does with their preferences:

57529563-81d21300-7335-11e9-99ad-403048b007ae

**Pull request merged**